src/components/Footer.vue

```diff
@@ -17,9 +17,27 @@
 
 <script lang="ts">
 import { Component, Vue } from 'vue-property-decorator';
+import fetch from 'node-fetch';
 
 @Component
 export default class Footer extends Vue {
+  public version = '';
+
+  readJson(): void {
+    // http://localhost:8080
+    fetch('https://kollavarsham.org/calendar-vue/version.json')
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP error ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((json) => {
+        this.version = json.text;
+      });
+  }
+
+  public runMethod = this.readJson()
```

Comment:
this might be a dumb question - but who calls this `runMethod` method? Is that more of a Vue.js component convention ?

Reply:
Never mind - I see it now!

Rather than this `runMethod` field/property, if we rename `readJson()` into `mounted()` (or call it from within another `mounted` callback), this will be more idiomatic in Vue.js

refer lifecycle hooks for a Vue component/view here: https://vuejs.org/v2/guide/instance.html#Instance-Lifecycle-Hooks

Reply:
@floydpink Yea, I couldn't find out how to do this with `vue-property-decorator` so I just left it at that, I'll trying doing it using lifecycle hooks